From 79d0493cdb6093a1d10cdfcecf6fb7bc220fa309 Mon Sep 17 00:00:00 2001
From: "qipeng.zha" <qipeng.zha@intel.com>
Date: Thu, 26 Mar 2015 16:53:53 +0800
Subject: [PATCH] Touch: fix access issue of debug sysfs node

a) debug_v2_enable sysfs node enabled read access but no read function
b) remove r/w access for other user, only grant for owner and group user

Change-Id: If92d2a715740837c5b22c3f45ee133bc021ca94b
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-26376
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-26379
Signed-off-by: qipeng.zha <qipeng.zha@intel.com>
---
 drivers/input/touchscreen/atmel_mxt_ts.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/input/touchscreen/atmel_mxt_ts.c b/drivers/input/touchscreen/atmel_mxt_ts.c
index e22db4d..5a3c799 100644
--- a/drivers/input/touchscreen/atmel_mxt_ts.c
+++ b/drivers/input/touchscreen/atmel_mxt_ts.c
@@ -509,7 +509,7 @@ static int mxt_debug_msg_init(struct mxt_data *data)
 {
 	sysfs_bin_attr_init(&data->debug_msg_attr);
 	data->debug_msg_attr.attr.name = "debug_msg";
-	data->debug_msg_attr.attr.mode = 0666;
+	data->debug_msg_attr.attr.mode = 0660;
 	data->debug_msg_attr.read = mxt_debug_msg_read;
 	data->debug_msg_attr.write = mxt_debug_msg_write;
 	data->debug_msg_attr.size = data->T5_msg_size * DEBUG_MSG_MAX;
@@ -3285,7 +3285,7 @@ static DEVICE_ATTR(hw_version, S_IRUGO, mxt_hw_version_show, NULL);
 static DEVICE_ATTR(object, S_IRUGO, mxt_object_show, NULL);
 static DEVICE_ATTR(update_fw, S_IWUSR, NULL, mxt_update_fw_store);
 static DEVICE_ATTR(update_cfg, S_IWUSR, NULL, mxt_update_cfg_store);
-static DEVICE_ATTR(debug_v2_enable, S_IWUSR | S_IRUSR, NULL,
+static DEVICE_ATTR(debug_v2_enable, S_IWUSR, NULL,
 		   mxt_debug_v2_enable_store);
 static DEVICE_ATTR(debug_notify, S_IRUGO, mxt_debug_notify_show, NULL);
 static DEVICE_ATTR(debug_enable, S_IWUSR | S_IRUSR, mxt_debug_enable_show,
-- 
1.9.1

